### DISCUSSION TITLE
Windows 11 Virtual Desktops

### DISCUSSION THREAD

**Discussion by BlueFinBima** [HeliosVirtualCockpit/Helios#567] [General]
I spent a couple of hours trying to work out how to stop CC being on all virtual desktops.  Failed to find a way.  Does anyone know a magic incantation to keep it on a single desktop, and any reason why it would be a bad thing to have it on a single virtual desktop?

**Comment by derammo** [comment]
Do you know what window style is causing the problem?   A simple example C# WPF program doesn't do this, does it?

**Reply by BlueFinBima** [reply]
I should have been more precise.  The main Control Center window behaves as you would expect, but the running profile is on all virtual desktops.

**Reply by BlueFinBima** [reply]
It became a problem for me when I was trying to use OBS to make a video.  It became an even bigger problem when I was debugging with a full screen profile and hit a breakpoint which I couldn't clear.

**Reply by derammo** [reply]
that makes more sense.    I was very surprised that the main Control Center window would do anything weird.     So instead, you can debug it by changing the code where it modifies the Windows Styles to make it "special-per-pixel-transparent" and "always-on-top."   Then you will know which window style causes the problem and then you can research / check MS docs.

**Reply by BlueFinBima** [reply]
Thanks.  Not sure that I've come across anywhere referring to "special-per-pixel-transparent".  I presumed that MonitorWindow.cs held all the relevant flag settings, but as far as I was able to tell, the settings used are necessary and benign.  I guess that the periodic "force to top" is causing the window to go to the top of the virtual desktop which is currently being displayed making it appear that the profile is on all displays.  If this is what is happening, then it is difficult to counter the current strategy without the CC being virtual desktop aware and installing the SDK etc which is more work than I am going to invest to address this niggle.

**Reply by derammo** [reply]
There is transparent and then there is the sort of transparent that lets you click through to the window underneath depending on the opaciy of what Helios (in this case) is drawing.  I believe Windows calls this per-pixel-transparency.   Anyway, Control Center has a prefs switch to turn off always-on-top so you should be able to trivially answer whether this is what is happening?  At least then we learned something and can document it instead of just being wasted time entirely.
